src/console/MeasurementCollectionToFhir/ProcessorStartup.cs

```diff
@@ -38,16 +41,31 @@ public void ConfigureServices(IServiceCollection services)
             services.Configure<ResourceIdentityOptions>(Configuration.GetSection("ResourceIdentity"));
             services.Configure<FhirClientFactoryOptions>(Configuration.GetSection("FhirClient"));
 
-            services.TryAddSingleton<IFactory<FhirClient>, FhirClientFactory>();
-            services.TryAddSingleton(sp => sp.GetRequiredService<IFactory<FhirClient>>().Create());
+            var url = new Uri(Environment.GetEnvironmentVariable("FhirService:Url"));
+            bool useManagedIdentity = Configuration.GetValue<bool>("FhirClient:UseManagedIdentity");
```

Comment:
Suggestion:  It would be good to wrap these in an options class or incorporate in it's own extension for easier reuse.

Reply:
Incorporated into ServiceCollectionExtensions.cs